pkg/ingress/group_loader.go

```diff
@@ -104,13 +115,52 @@ func (m *defaultGroupLoader) Load(ctx context.Context, groupID GroupID) (Group,
 	}, nil
 }
 
-// matchesIngressClass tests whether Ingress matches ingress class of this group manager.
-func (m *defaultGroupLoader) matchesIngressClass(ing *networking.Ingress) bool {
-	ingClass := ing.Annotations[annotations.IngressClass]
-	if m.ingressClass == "" {
-		return ingClass == "" || ingClass == defaultIngressClass
+// matchesIngressClass tests whether provided Ingress are matched by this group loader.
+func (m *defaultGroupLoader) matchesIngressClass(ctx context.Context, ing *networking.Ingress) (bool, error) {
+	var matchesIngressClassResults []bool
+	if ingClassAnnotation, exists := ing.Annotations[annotations.IngressClass]; exists {
+		matchesIngressClass := m.matchesIngressClassAnnotation(ctx, ingClassAnnotation)
+		matchesIngressClassResults = append(matchesIngressClassResults, matchesIngressClass)
+	}
+
+	if ing.Spec.IngressClassName != nil {
+		matchesIngressClass, err := m.matchesIngressClassName(ctx, *ing.Spec.IngressClassName)
+		if err != nil {
+			return false, err
+		}
+		matchesIngressClassResults = append(matchesIngressClassResults, matchesIngressClass)
+	}
+
+	if len(matchesIngressClassResults) == 2 {
+		if matchesIngressClassResults[0] != matchesIngressClassResults[1] {
+			m.eventRecorder.Event(ing, corev1.EventTypeWarning, k8s.IngressEventReasonConflictIngressClass, "conflict IngressClass by `spec.IngressClass` and `kubernetes.io/ingress.class` annotation")
```

Comment:
nit: lets reword the error message
conflicting values for IngressClass ..

Reply:
thanks,  addressed it :D